### DISCUSSION TITLE
Pass Tool Use Context to Title Generation

### DISCUSSION THREAD

**Discussion by ekatiyar** [open-webui/open-webui#4508] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently title generation is based only on the user's initial input prompt. While this works for most cases, it isn't as effective when the prompt requires the model to use a tool (e.g. webscraping) in order to obtain a more full context to answer the prompt. An example use case would be summarizing a webpage - in this case the model hallucinates the contents of the webpage when generating a title, often resulting in a totally off-topic title.

**Describe the solution you'd like**
Provide the context/output of any tool calls resulting from the initial prompt before generating the title. This will result in better, more informed conversation titles.

**Describe alternatives you've considered**
Another option is to wait till the model has sent an initial reply in the conversation, like what OpenAI does in their ChatGPT interface, but this is largely unnecessary for most conversations without tool use, and just including tool outputs should get us most of the way there in terms of correctness, and would maintain current performance for non-tool conversations.

**Comment by oznav2** [comment]
this issue is true even when an attaching a document and asking the model a summary. the title in such case should be based on the context of the document rather than on the user initial request within the prompt itself
